client/src/query_builder.rs

```diff
@@ -0,0 +1,58 @@
+use std::fmt::Debug;
+
+use iroha_data_model::{
+    predicate::PredicateBox,
+    query::{sorting::Sorting, Pagination, Query},
+    Value,
+};
+
+use crate::client::{Client, QueryOutput, QueryResult};
+
+pub struct QueryRequestBuilder<'a, R> {
+    client: &'a Client,
+    request: R,
+    pagination: Pagination,
+    filter: PredicateBox,
+    sorting: Sorting,
+}
+
+impl<'a, R> QueryRequestBuilder<'a, R>
+where
+    R: Query + Debug,
+    R::Output: QueryOutput,
+    <R::Output as TryFrom<Value>>::Error: Into<eyre::Error>,
+{
+    pub(crate) fn new(client: &'a Client, request: R) -> Self {
+        Self {
+            client,
+            request,
+            pagination: Pagination::default(),
+            sorting: Sorting::default(),
+            filter: PredicateBox::default(),
+        }
+    }
+
+    pub fn filter(mut self, filter: PredicateBox) -> Self {
```

Comment:
Within Iroha codebase, builders usually have `with_` form. I think it would be more consistent to use `with_filter`, `with_sorting` and `with_pagination`.